libyul/AsmAnalysis.cpp

```diff
@@ -417,24 +417,30 @@ std::vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 				std::string functionName = _funCall.functionName.name.str();
 				if (functionName == "datasize" || functionName == "dataoffset")
 				{
-					if (!m_dataNames.count(std::get<Literal>(arg).value))
+					auto const& lit = std::get<Literal>(arg);
+					if (!m_dataNames.count(YulString(lit.value.hint() ? *lit.value.hint() : formatLiteral(lit))))
```

Comment:
We have `_literal` and `lit` in the same context. This naming is ambiguous. It's much better when names require less context to be understood.
```suggestion
					auto const& argumentAsLiteral = std::get<Literal>(arg);
					if (!m_dataNames.count(YulString(argumentAsLiteral.value.hint() ? *argumentAsLiteral.value.hint() : formatLiteral(argumentAsLiteral))))
```